### PR TITLE
feat(DLI): add new resource: flink job

### LINKED ISSUE #1654: [DLI]Need a new resource to manage flink SQL jobs
For DLI service, we need a new resource to manage flink SQL jobs

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
feat(DLI): add new resource: flink job

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1654 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccResourceDliFlinkJob'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccResourceDliFlinkJob -timeout 360m -parallel 4
=== RUN   TestAccResourceDliFlinkJob_basic
=== PAUSE TestAccResourceDliFlinkJob_basic
=== CONT  TestAccResourceDliFlinkJob_basic
--- PASS: TestAccResourceDliFlinkJob_basic (68.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       68.212s
```

### CODE REVIEW

File: docs/resources/dli_flinksql_job.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  type       = "flink_sql_job"
+  sql_body   = var.sql
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the MapReduce cluster resource. If omitted, the
```
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: s/MapReduce cluster resource/flink sql job/g

File: docs/resources/dli_flinksql_job.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+* `run_mode` - (Optional, String) Specifies job running mode. The options are as follows:
+
+  + **shared_cluster**: indicates that the job is running on a shared cluster.
+  + **exclusive_cluster**: indicates that the job is running on an exclusive cluster.
+  + **edge_node**: indicates that the job is running on an edge node.
+  
+  The default value is `shared_cluster`.
+
+* `desc` - (Optional, String) Specifies job description. Length range: 1 to 512 characters.
```
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: please rename to `description`

File: docs/resources/dli_flinksql_job.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+* `queue_name` - (Optional, String) Specifies name of a queue.
+
+* `sql_body` - (Optional, String) Specifies stream SQL statement, which includes at least the following
```
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: suggest to rename to `sql`

File: docs/resources/dli_flinksql_job.md (hunk 2 above)
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: please rename to `description`

File: docs/resources/dli_flinksql_job.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ three parts: source, query, and sink. Length range: 1024x1024 characters.
+
+* `cu_number` - (Optional, Int) Specifies number of CUs selected for a job. The default value is 2.
+
+* `parallel_number` - (Optional, Int) Specifies number of parallel for a job. The default value is 1.
+
+* `checkpoint_enabled` - (Optional, Boolen) Specifies whether to enable the automatic job snapshot function.
```
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: rename to `enable_checkpoint`?

File: docs/resources/dli_flinksql_job.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  The default value is 1.
+
+* `checkpoint_interval` - (Optional, Int) Specifies Snapshot interval. The unit is second.
+  The default value is 10.
+
+* `obs_bucket` - (Optional, String) Specifies OBS path. OBS path where users are authorized to save the
+  snapshot. This parameter is valid only when `checkpoint_enabled` is set to `true`. OBS path where users are authorized
+  to save the snapshot. This parameter is valid only when `log_enabled` is set to `true`.
+
+* `log_enabled` - (Optional, Boolean) Specifies whether to enable the function of uploading job logs to
```
Comment (ShiChangkuo) [inline docs/resources/dli_flinksql_job.md]: s/Boolean/Bool/g

File: huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go [hunk 6]
```diff
@@ -0,0 +1,599 @@
...
+		d.Set("dirty_data_strategy", detail.JobConfig.DirtyDataStrategy),
+		d.Set("udf_jar_url", detail.JobConfig.UdfJarUrl),
+		d.Set("manager_cu_number", detail.JobConfig.ManagerCuNumber),
+		d.Set("tm_cus", detail.JobConfig.TmCus),
+		d.Set("tm_slot_num", detail.JobConfig.TmSlotNum),
+		d.Set("resume_checkpoint", detail.JobConfig.ResumeCheckpoint),
+		d.Set("resume_max_num", detail.JobConfig.ResumeMaxNum),
+		d.Set("runtime_config", utils.TagsToMap(parseConfig(detail.JobConfig.RuntimeConfig))),
+		d.Set("status", detail.Status),
+		// d.Set("tags", utils.TagsToMap(detail.t)), //没有返回
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_flinksql_job.go]: you can delete this line directly

File: docs/resources/dli_flinksql_job.md (hunk 4 above)
Comment (hmbig2) [reply docs/resources/dli_flinksql_job.md]: keep it as it is

**Pull request merged**